samples/quantum-approximation-optimization/qaoa-for-quadratic-unconstrained-binary-optimization.ipynb

```diff
@@ -0,0 +1,820 @@
+    "    }\n",
+    "\n",
+    "    try:\n",
+    "        res = minimize(\n",
+    "            func,\n",
+    "            args=Q,\n",
+    "            x0=np.r_[guess[\"t\"], guess[\"s\"]],\n",
+    "            method=\"Nelder-Mead\",\n",
+    "            tol=1e-5,\n",
+    "            options={\"maxiter\": 10},\n",
+    "        )\n",
+    "        scores.append(res.fun)\n",
+    "        params.append(res.x)\n",
+    "    except Exception as e:\n",
+    "        pass"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now plot the sample that we woud obtain using the optimal variational parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "optimal_count_dict = quantum_loop(params[np.argmin(scores)])\n",
+    "plot_distribution(optimal_count_dict)"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "QAOA is capable of finding good variational parameters $\\tau$ and $t$. Now, sampling from this final state $|\\psi(t_{f})\\rangle$ will return both optimal strings with high probability."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "However, using QAOA to solve the problem is not the best idea; it's difficult to yield a >90% quality solution without going to high depths of the QAOA, implying that the growing closed-loop optimization can rapidly become expensive, with no guarantee of convergence. We therefore propose another approach called the Quantum Adiabatic Algorithm (QAA). This fast, reliant and exclusively analog method shows optimal convergence to the solution."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Quantum Adiabatic Algorithm"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The idea behind the adiabatic algorithm (see [Albash, Lidar, 2018](https://arxiv.org/pdf/1611.04471.pdf)) is to slowly evolve the system from an easy-to-prepare groundstate to the groundstate of $H_Q$. If done slowly enough, the system of atoms stays in the instantaneous ground-state.\n",
+    "\n",
+    "In our case, we continuously vary the parameters $\\Omega(t), \\delta(t)$ in time, starting with $\\Omega(0)=0, \\delta(0)<0$ and ending with $\\Omega(0)=0, \\delta>0$. The ground-state of $H(0)$ corresponds to the initial state $|00000\\rangle$ and the ground-state of $H(t_f)$ corresponds to the ground-state of $H_Q$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": "The Rydberg blockade radius is directly linked to the Rabi frequency $\\Omega$ and is obtained using `DigitalAnalogDevice.rydberg_blockade_radius()`. In this notebook, $\\Omega$ is initially fixed to a frequency of 1 rad/µs. We can therefore build the adjacency matrix $A$ of $G$ in the following way:"
```

Comment:
Do you really want to use DigitalAnalogDevice ? I think AnalogDevice would suit better what you are trying to prove.

Reply:
The sample I received from Anne-Claire was using Chadoq2, which was triggering the following warning in the console:

>DeprecationWarning: The 'Chadoq2' device has been deprecated. For a similar device combining global and local addressing, consider using `DigitalAnalogDevice`.

So I applied that suggestion :grinning:

Since I'm not the author of the notebook (just integrating it here), I suppose you are more knowledgable than me @a-corni.

Your call.

Reply:
It's okay, let's go with this solution :)